src/coreclr/jit/emitriscv64.cpp

```diff
@@ -4905,8 +4912,7 @@ regNumber emitter::emitInsTernary(instruction ins, emitAttr attr, GenTree* dst,
                     {
                         if (attr == EA_4BYTE)
                         {
-                            emitIns_R_R_I(INS_slli, EA_8BYTE, dstReg, dstReg, 32);
-                            emitIns_R_R_I(INS_srli, EA_8BYTE, dstReg, dstReg, 32);
+                            emitIns_R_R(INS_sext_w, EA_8BYTE, dstReg, dstReg);
```

Comment:
Previous code was zero-extending, not sign-extending. Intended?

Also, why in multiplication? The PR description says only add/subtract had a problem.

Reply:
Not intended change. Revoked